src/BizHawk.Client.Common/movie/tasproj/IStateManagerSettings.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace BizHawk.Client.Common
+{
+	[JsonConverter(typeof(StateManagerSettingsConverter))]
+	public interface IStateManagerSettings
+	{
+		IStateManager CreateManager(Func<int, bool> reserveCallback);
+
+		IStateManagerSettings Clone();
+	}
+
+	// Legacy deserialization support. It's ugly, but Newtonsoft is dumb like that.
+	// All classes that implement IStateManager MUST use NoConverter as their converter or face infinite recursion.
+	public class StateManagerSettingsConverter : JsonConverter<IStateManagerSettings>
+	{
+		public override IStateManagerSettings ReadJson(JsonReader reader, Type objectType, IStateManagerSettings existingValue, bool hasExistingValue, JsonSerializer serializer)
+		{
+			var jsonObject = Newtonsoft.Json.Linq.JObject.Load(reader);
+			if (jsonObject.TryGetValue("$type", out JToken value))
+			{
+				string type = (string)value;
+				if (type.Contains(nameof(PagedStateManager.PagedSettings)))
+					return jsonObject.ToObject<PagedStateManager.PagedSettings>(serializer);
+				else if (type.Contains(nameof(ZwinderStateManagerSettings)))
+					return jsonObject.ToObject<ZwinderStateManagerSettings>(serializer);
+				else
+					throw new Exception("Invalid state manager type."); // This will get eaten.
+			}
+			else
+			{
+				if (reader.Path.Contains("DefaultTasStateManagerSettings"))
+				{
+					// We want to default to the new manager.
+					// Users will only keep the zwinder manager as their default if they specify it after we've added the paged manager.
+					return new PagedStateManager.PagedSettings();
+				}
+				else
+				{
+					// But inside a movie file, keep existing settings.
+					return jsonObject.ToObject<ZwinderStateManagerSettings>(serializer);
+				}
+			}
+		}
+
+		public override bool CanWrite => false;
+
+		public override void WriteJson(JsonWriter writer, IStateManagerSettings value, JsonSerializer serializer) => throw new NotImplementedException();
+	}
+
+	// Newtonsoft makes classes inherit converters from implemented interfaces. Using this converter undoes that.
```

Comment:
Are you able to set `[JsonConverter]` on `IStateManager.Settings` directly?

Reply:
That would do nothing even if it does work, because we never serialize state managers.
It would be possible to get the same results by modifying the code that does the serialization (with some serializer settings), but I thought doing it here would be better. That maybe is not actually the case given how convoluted it makes things.
But I was also thinking we'd probably eventually remove the old one, or at least drop support for deserializing things that don't contain type info. In that case, we would just delete the converter and referencing attributes when that happens.